main/src/ca/uwaterloo/flix/language/phase/unification/Unification.scala

```diff
@@ -280,8 +280,29 @@ object Unification {
         case Result.Err(err: UnificationError.MultipleMatchingInstances) =>
           throw InternalCompilerException(s"Unexpected unification error: $err")
       }
+    })
+  }
+
+  /**
+    * Unifies the `expected` type with the `actual` type.
+    */
+  def expectTypeM(expected: Type, actual: Type, loc: SourceLocation)(implicit flix: Flix): InferMonad[Type] = {
+    def handler(e: TypeError): TypeError = e match {
+      case e: TypeError.MismatchedTypes => TypeError.UnexpectedType(expected, actual, loc)
+      case e => e
     }
-    )
+
+    unifyTypeM(expected, actual, loc).transformError(handler)
+  }
+
+  /**
+    * Unifies the `expected` type with the `actual` type (and unifies `bind` with the result).
+    */
+  def expectTypeM(expected: Type, actual: Type, bind: Type.Var, loc: SourceLocation)(implicit flix: Flix): InferMonad[Type] = {
+    for {
+      r <- expectTypeM(expected, actual, loc)
+      _ <- unifyTypeM(bind, r, loc)
```

Comment:
To be sure I understand: We can throw away the second unification result because it's a var-to-type binding, and the var won't refine the type at all.